pygmt/src/triangulate.py

```diff
@@ -0,0 +1,381 @@
+"""
+triangulate - Delaunay triangulation or Voronoi partitioning and gridding of
+Cartesian data.
+"""
+import warnings
+
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+from pygmt.io import load_dataarray
+
+
+class triangulate:  # pylint: disable=invalid-name
+    """
+    Delaunay triangulation or Voronoi partitioning and gridding of Cartesian
+    data.
+
+    Triangulate reads in x,y[,z] data and performs Delaunay triangulation,
+    i.e., it finds how the points should be connected to give the most
+    equilateral triangulation possible. If a map projection (give ``region``
+    and ``projection``) is chosen then it is applied before the triangulation
+    is calculated. By default, the output is triplets of point id numbers that
+    make up each triangle. The id numbers refer to the points position (line
+    number, starting at 0 for the first line) in the input file. If ``outgrid``
+    and ``spacing`` are set a grid will be calculated based on the surface
+    defined by the planar triangles. The actual algorithm used in the
+    triangulations is either that of Watson [1982] or Shewchuk [1996] [Default
+    is Shewchuk if installed; type ``gmt get GMT_TRIANGULATE`` on the command
+    line to see which method is selected]. Furthermore, if the Shewchuk
+    algorithm is installed then you can also perform the calculation of Voronoi
+    polygons and optionally grid your data via the natural nearest neighbor
+    algorithm. **Note**: For geographic data with global or very large extent
```

Comment:
```suggestion
    algorithm.

    Notes
    -----
    For geographic data with global or very large extent
```
Should this go in a separate section following https://numpydoc.readthedocs.io/en/latest/format.html#notes?

Reply:
Good idea, now it looks like this:

![image](https://user-images.githubusercontent.com/23487320/158075833-9f1d39a3-0b99-4c0b-b76d-2b6e34b645f4.png)